homeassistant/components/sunweg/__init__.py

```diff
@@ -29,9 +19,11 @@ async def async_setup_entry(
     api = APIHelper(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
     if not await hass.async_add_executor_job(api.authenticate):
         raise ConfigEntryAuthFailed("Username or Password may be incorrect!")
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = SunWEGData(
-        api, entry.data[CONF_PLANT_ID]
+    coordinator = SunWEGDataUpdateCoordinator(
+        hass, api, entry.data[CONF_PLANT_ID], entry.data[CONF_NAME]
     )
+    await coordinator.async_config_entry_first_refresh()
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```

Comment:
Store this in runtime data instead for the config entry
https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry

Reply:
Hey @rokam, the runtime-data is a really nice new feature that's been introduced since we did the review in April.  @gjohansson-ST is a core member so hopefully you're really close to merge with this PR now.  Good luck!